File: BaseLib/src/main/java/com/hjc/baselib/http/observer/BaseCommonObserver.java
package com.hjc.baselib.http.observer;

import com.blankj.utilcode.util.ToastUtils;

/**
 * @Author: HJC
 * @Date: 2019/1/7 11:52
 * @Description: 通用的Observer
 */
public abstract class BaseCommonObserver<T> extends BaseObserver<T> {

    @Override
    public void onFailure(String errorMsg) {
        ToastUtils.showShort(errorMsg);
    }

}

File: library-net/src/main/java/com/hjc/library_net/observer/BaseCommonObserver.kt
package com.hjc.library_net.observer

import com.blankj.utilcode.util.ToastUtils
import com.hjc.library_base.viewmodel.BaseViewModel
import com.hjc.library_net.exception.ExceptionUtils.handleException

/**
 * @Author: HJC
 * @Date: 2019/1/7 11:52
 * @Description: 通用的Observer
 */
abstract class BaseCommonObserver<T>(baseViewModel: BaseViewModel) :
    BaseObserver<T>(baseViewModel) {

    override fun onFailure(e: Throwable) {
        val errorMsg = handleException(e)
        ToastUtils.showShort(errorMsg)
    }
}